crates/common/src/utils/position_utils.rs

```diff
@@ -0,0 +1,75 @@
+use crate::model::position::Position;
+use bstr::BStr;
+use bstr::ByteSlice;
+
+/// Get position of an offset in a code and return a [[Position]].
+pub fn get_position_in_string(content: &str, offset: usize) -> anyhow::Result<Position> {
+    let bstr = BStr::new(&content);
+
+    let mut line_number: u32 = 1;
+
+    for line in bstr.lines_with_terminator() {
+        let start_index = line.as_ptr() as usize - content.as_ptr() as usize;
+        let end_index = start_index + line.len();
+
+        if offset >= start_index && offset < end_index {
+            let mut col_number: u32 = 1;
+            for grapheme in line.grapheme_indices() {
+                // It's exactly the index we are looking for.
+                if offset == start_index + grapheme.0 {
+                    return Ok(Position {
+                        line: line_number,
+                        col: col_number,
+                    });
+                }
+
+                // The offset is within the grapheme we are looking for, it's the next col.
+                if offset >= start_index + grapheme.0 && offset < start_index + grapheme.1 {
+                    return Ok(Position {
+                        line: line_number,
+                        col: col_number + 1,
```

Comment:
It might be nicer to destructure the iterator value, since using `.0` and `.1` isn't clear on what it actually is:

```suggestion
            for (grapheme_start, grapheme_end, _) in line.grapheme_indices() {
                // It's exactly the index we are looking for.
                if offset == start_index + grapheme_start {
                    return Ok(Position {
                        line: line_number,
                        col: col_number,
                    });
                }

                // The offset is within the grapheme we are looking for, it's the next col.
                if offset > start_index + grapheme_start && offset < start_index + grapheme_end {
                    return Ok(Position {
                        line: line_number,
                        col: col_number + 1,
```

Also minor nit, we don't need `>=` and can use `>` to check if the offset is greater, we already checked if it's equal above.

Reply:
Thank you for pointing this out, great idea to destructure!
I addressed both feedback!